plugins/device/rest.go

```diff
@@ -170,6 +170,42 @@ func deviceEditHandler(w http.ResponseWriter, r *http.Request) {
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, nil)
 }
 
+func deviceDeleteHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
sorry, I missed it earlier. We need to remove the device using `vgremove` and `pvremove` as a transaction step. Deleting device from etcd should be last step.